includes/class-federated-reactions-settings.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Federated Reactions Settings file.
+ *
+ * @package ActivityPub
+ */
+
+namespace Activitypub;
+
+/**
+ * Federated Reactions Settings class.
+ */
+class Federated_Reactions_Settings {
+
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+		add_action( 'init', array( self::class, 'register_post_meta' ), 11 );
+		add_action( 'admin_init', array( self::class, 'register_settings' ) );
+	}
+
+	/**
+	 * Register post meta for federated reactions.
+	 */
+	public static function register_post_meta() {
+		$ap_post_types = get_post_types_by_support( 'activitypub' );
+		foreach ( $ap_post_types as $post_type ) {
+			register_post_meta(
+				$post_type,
+				'activitypub_reactions_enabled',
+				array(
+					'show_in_rest'      => true,
+					'single'            => true,
+					'type'              => 'string',
+					'sanitize_callback' => function ( $value ) {
+						return in_array( $value, array( '0', '1' ), true ) ? $value : '1';
+					},
+				)
+			);
+		}
+	}
+
+	/**
+	 * Register settings.
+	 */
+	public static function register_settings() {
+		register_setting(
+			'discussion',
```

Comment:
@obenland I like the integration on the discussion page, but I would also add the setting to the ActivityPub settings. I would assume that users will first search in the plugins settings.